src/main/java/org/spongepowered/common/mixin/core/entity/player/EntityPlayerMixin.java

```diff
@@ -391,7 +401,7 @@ private void spongePlaySound(final World world, final EntityPlayer player, final
 
     @Override
     public boolean bridge$affectsSpawning() {
-        return this.affectsSpawning && !this.isSpectator();
+        return this.affectsSpawning && !this.isSpectator() && !this.bridge$isVanished();
```

Comment:
[If a player is targetable becuase `VANISH_PREVENTS_TARGETING` is `false`](https://github.com/SpongePowered/SpongeAPI/blob/stable-7/src/main/java/org/spongepowered/api/data/key/Keys.java#L2271), should spawning also be unaffected?

Reply:
good point, I'll change that